internal/common/pid_sock_status.go

```diff
@@ -42,19 +52,51 @@ func WriteAndHandleStatusFile() error {
 	if err != nil {
 		return fmt.Errorf("Failed to generate a random number for the unix socket path: '%w'.", err)
 	}
-	statusJson.UnixSocketPath = fmt.Sprintf("/tmp/autograder-%s.sock", unixFileNumber)
+	statusJson.UnixSocketPath = filepath.Join("/tmp", fmt.Sprintf("autograder-%s.sock", unixFileNumber))
 
 	err = util.ToJSONFile(statusJson, statusPath)
 	if err != nil {
-		return fmt.Errorf("Failed to write the pid to the status file: '%w'.", err)
+		return fmt.Errorf("Failed to write to the status file: '%w'.", err)
 	}
 
-	return nil
+	return err
 }
 
-func GetUnixSocketPath() (string, error) {
-	Lock(PID_SOCK_LOCK)
-	defer Unlock(PID_SOCK_LOCK)
+func checkAndHandleStalePid() (bool, error) {
+	statusPath := GetStatusPath()
+
+	if !util.IsFile(statusPath) {
+		return true, nil
+	}
+
+	var statusJson StatusInfo
+
+	err := util.JSONFromFile(statusPath, &statusJson)
+	if err != nil {
+		return false, fmt.Errorf("Failed to read the status file: '%w'.", err)
+	}
+
+	process, _ := os.FindProcess(statusJson.Pid)
```

Comment:
Make a function (isAlive()?) that handles this.
It's not a lot of code, but it's confusing enough that it would be good to box it off.
Then you can also include a function-level comment on it.